### PR TITLE
feat: add charmed_zookeeper plugin

### PR DESCRIPTION
**Pull request by marcoppenheimer**
Code is mostly self-explanatory.

To test oneself on an LXD cloud:

```bash
juju add-model temp
juju deploy zookeeper --channel edge -n 3
juju deploy kafka --channel edge -n 3
juju relate kafka zookeeper
juju deploy kafka-test-app --series=jammy --channel=edge --config role=producer --config topic_name="HOT-TOPIC" --config num_messages=100000 --force producer
juju deploy kafka-test-app --series=jammy --channel=edge --config role=consumer --config topic_name="HOT-TOPIC" --config num_messages=100000 --force consumer
juju relate kafka producer
juju relate kafka consumer

# wait for active/idle

juju ssh zookeeper/0 sudo -i

# in SSH session

git clone https://github.com/marcoppenheimer/sos.git
cd sos
git checkout feat/charmed_zookeeper
python3 setup.py install
./bin/sos report --only-plugins "charmed_zookeeper"
cd /tmp
tar xvf *.xz

# inspect the actual outputted files as you wish

ls -lah ./**/var/snap/charmed-zookeeper/current/etc/zookeeper
ls -lah ./**/var/snap/charmed-zookeeper/common/var/log/zookeeper
ls -lah ./**/sos_commands/charmed_zookeeper
```

### CODE REVIEW

File: sos/report/plugins/charmed_zookeeper.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            ) or file_dt > datetime.strptime(
+                str(self.get_option("date-to")), DATE_FORMAT
+            ):
+                # skip files outside given range
+                self.add_forbidden_path(file)
+
+        # hide keys/stores
+        self.add_forbidden_path(
+            [
+                f"{PATHS['CONF']}/*.pem",
```
Comment (zmraul) [inline sos/report/plugins/charmed_zookeeper.py:99]: I think `/*.key` as well?

**Review by zmraul — CHANGES_REQUESTED** [review]

File: sos/report/plugins/charmed_zookeeper.py (hunk 1 above)
Comment (marcoppenheimer) [reply sos/report/plugins/charmed_zookeeper.py:99]: Ah, same for Kafka too. I'll add.

**Comment by deusebio** [comment]
Ops, this did not showed up on my review list :(

I'm very happy with the changes. Great that keys were excluded. Are they excluded for Kafka as well already?

**Comment by deusebio** [comment]
LGTM

**Pull request merged**